video/image_writer.c

```diff
@@ -229,9 +229,10 @@ static bool write_lavc(struct image_writer_ctx *ctx, mp_image_t *image, const ch
     ret = avcodec_receive_packet(avctx, pkt);
     if (ret < 0)
         goto error_exit;
-    success = true;
 
     fwrite(pkt->data, pkt->size, 1, fp);
+    fflush(fp);
+    success = !ferror(fp);
```

Comment:
Since `fclose`'s return value is used, `fflush` is probably redundant here, but `ferror` is not.

`fclose` will flush the buffer and report errors for the flush caused by `fclose`. But an error might have occurred earlier - in which case the error indicator will be set and `ferror` checks for that.

In other words, I don't believe `fclose` reports whether the error indicator was set or not.

Reply:
Also, now that I think about it, since `fwrite` is the only IO function used here - checking for it's return value should work. Any remaining buffered data will flushed by `fclose` (and any error during `fclose`s flush will be reported by it).